python/tank/platform/software_launcher.py

```diff
@@ -234,6 +234,21 @@ def shotgun(self):
         """
         return self.sgtk.shotgun
 
+    def _get_minimum_supported_version(self):
+        """
+        The minimum software version that is supported by the launcher.
+        Returned as a string, for example "2015" or "2015.3.sp3".
+        Returns ``None`` if no constraint has been set.
+        Also see related method :meth:`~is_version_supported`.
+        """
+        return self.__minimum_supported_version
+
+    def _set_minimum_supported_version(self, version):
```

Comment:
Hey JF! the setter isn't there for the tests alone - it's meant to be used like this: 

https://github.com/shotgunsoftware/tk-photoshopcc/blob/7ddddf373d9c150883dbb706b6decd1027b06d40/startup.py#L33

As a simple, pragmatic way for an engine implementation of the launcher to optionally implement a minimum version. 

We could also implement it as a protected method on the software launcher that the various engine implementations could call, `self._set_minimum_version("2016)`, but it just seemed overkill. What do you think of the photoshop implementation above?